cpp/Cipher/MGLRsa.cpp

```diff
@@ -177,7 +177,7 @@ std::pair<StringOrBuffer, StringOrBuffer> generateRSAKeyPair(
                               "Failed to encode public and/or private key");
   }
 
-  return std::make_pair(publicBuffer.value(), privateBuffer.value());
+  return {publicBuffer.value(), privateBuffer.value()};
```

Comment:
@Szymon20000 @ospfranco sorry to ping you, but I'm stuck here.  Trying to switch over to use std::variant and std::pair here, and getting `No matching constructor for initialization of 'std::pair<JSVariant, JSVariant>' (aka 'pair<variant<std::nullptr_t, bool, int, double, long, long long, basic_string<char>, margelo::ByteSource>, variant<std::nullptr_t, bool, int, double, long, long long, basic_string<char>, margelo::ByteSource>>')` - also getting a similar error elsewhere.  I should be self-sufficient if I can figure out at least one of these `crypto.subtle` API functions fully with tests.

Reply:
you first need to create a variant with the buffer value before creating the pair. The compiler doesn't know how to create a variant from the buffer value on its own.

Reply:
I thought I had done that here: https://github.com/boorad/react-native-quick-crypto/blob/more-subtle-impls/cpp/MGLKeys.cpp#L251-L262 as a result of `WritePrivateKey` and `WritePublicKey`.  I need to do it again here?  Both sides of the `pair`, when calling `.value()` supposedly result in `JSVariant`.  Not sure about `const` or `&`, though.

This results in the same error:
```c++
  const JSVariant& publicKey = publicBuffer.value();
  const JSVariant& privateKey = privateBuffer.value();
  return {publicKey, privateKey};
```

Reply:
ok, I was mistakenly calling the copy constructor for `ByteSource` values in the `variant`.  When I surround the two `pair` elements with `std::move()` the move constructors are used, and all is well.